src/Cons.elm

```diff
@@ -194,81 +266,96 @@ foldr1 f c =
 
     fromList [] == Nothing
     fromList [1, 2, 3] == Just <| cons 1 [2, 3]
 -}
 fromList : List a -> Maybe (Cons a)
 fromList l =
-  case l of
-    [] -> Nothing
-    head::tail -> Just <| cons head tail
+    case l of
+        [] ->
+            Nothing
+
+        head :: tail ->
+            Just <| cons head tail
+
 
 {-| A cons with the given head and tail.
 
-    c = cons' "a" Nothing
+    c = consWithMaybe "a" Nothing
     toList c == ["a"]
 
-    d = cons' 1 <| Just <| cons' 2 <| Just <| cons' 3 Nothing
+    d = consWithMaybe 1 <| Just <| consWithMaybe 2 <| Just <| consWithMaybe 3 Nothing
     toList d = [1, 2, 3]
 -}
-cons' : a -> Maybe (Cons a) -> Cons a
-cons' head tail = cons head <| toList' tail
+consWithMaybe : a -> Maybe (Cons a) -> Cons a
+consWithMaybe head tail =
+    cons head <| toMaybeList tail
+
 
 {-| The head and tail of the cons.
 
-    c = cons' "a" Nothing
-    uncons' c == ("a", Nothing)
+    c = consWithMaybe "a" Nothing
+    unconsWithMaybe c == ("a", Nothing)
 
-    d = cons' 1 <| Just <| cons' 2 <| Just <| cons' 3 Nothing
-    uncons' d == (1, Just <| cons' 2 <| Just <| cons' 3 Nothing)
+    d = consWithMaybe 1 <| Just <| consWithMaybe 2 <| Just <| consWithMaybe 3 Nothing
+    unconsWithMaybe d == (1, Just <| consWithMaybe 2 <| Just <| consWithMaybe 3 Nothing)
 
     maximum : Cons comparable -> comparable
     maximum c =
-      case uncons' c of
+      case unconsWithMaybe c of
         (first, Nothing) -> first
         (first, Just rest) -> max first <| maximum rest
 -}
-uncons' : Cons a -> (a, Maybe (Cons a))
-uncons' (Cons head tail) = (head, fromList tail)
+unconsWithMaybe : Cons a -> ( a, Maybe (Cons a) )
+unconsWithMaybe (Cons head tail) =
+    ( head, fromList tail )
+
 
 {-| The tail of the cons.
 
-    c = cons' "a" Nothing
-    tail' c == Nothing
+    c = consWithMaybe "a" Nothing
+    maybeTail c == Nothing
 
-    d = cons' 1 <| Just <| cons' 2 <| Just <| cons' 3 Nothing
-    tail' d == Just <| cons' 2 <| Just <| cons' 3 Nothing
+    d = consWithMaybe 1 <| Just <| consWithMaybe 2 <| Just <| consWithMaybe 3 Nothing
+    maybeTail d == Just <| consWithMaybe 2 <| Just <| consWithMaybe 3 Nothing
 
     length : Cons a -> Int
     length c =
-      case tail' c of
+      case maybeTail c of
         Nothing -> 1
         Just rest -> 1 + length rest
 -}
-tail' : Cons a -> Maybe (Cons a)
-tail' = tail >> fromList
+maybeTail : Cons a -> Maybe (Cons a)
+maybeTail =
+    tail >> fromList
+
 
 {-| Convert the cons to the equivalent list, or the empty list for Nothing.
 
 This is the inverse of fromList.
 
     c = fromList []
     c == Nothing
-    toList' c == []
+    toMaybeList c == []
 
     c = fromList [1, 2, 3]
     c == Just <| cons 1 [2, 3]
-    toList' c == [1, 2, 3]
+    toMaybeList c == [1, 2, 3]
 -}
-toList' : Maybe (Cons a) -> List a
-toList' = Maybe.map toList >> Maybe.withDefault []
+toMaybeList : Maybe (Cons a) -> List a
+toMaybeList =
```

Comment:
This name makes it sound like `Maybe (List a)` should be the return type. But `maybeToList` isn't ideal either… tricky!